aclk/aclk_query.c

```diff
@@ -115,20 +115,20 @@ static int http_api_v2(struct aclk_query_thread *query_thr, aclk_query_t query)
         char *node_uuid = query->data.http_api_v2.query + strlen(NODE_ID_QUERY);
         char nodeid[UUID_STR_LEN];
         if (strlen(node_uuid) < (UUID_STR_LEN - 1)) {
-            error("URL requests node_id but there is not enough chars following. Returning 404 to Cloud.");
+            error_report(CLOUD_EMSG_MALFORMED_NODE_ID);
             retval = 1;
             w->response.code = 404;
-            aclk_http_msg_v2_err(query_thr->client, query->callback_topic, query->msg_id, w->response.code, NULL, 0);
+            aclk_http_msg_v2_err(query_thr->client, query->callback_topic, query->msg_id, w->response.code, CLOUD_EC_MALFORMED_NODE_ID, CLOUD_EMSG_MALFORMED_NODE_ID, NULL, 0);
             goto cleanup;
         }
         strncpyz(nodeid, node_uuid, UUID_STR_LEN - 1);
 
         query_host = node_id_2_rrdhost(nodeid);
         if (!query_host) {
-            error("Host with node_id \"%s\" not found! Returning 404 to Cloud!", node_uuid);
+            error_report("Host with node_id \"%s\" not found! Returning 404 to Cloud!", node_uuid);
```

Comment:
The `node_uuid` is not actually the node id. It is the entire query starting with the node id. So the error message ends up logging something like: 
```
Host with node_id "67c22c43-b604-40fc-b60d-4dc53f83edef/api/v1/data?after=1647773828&before=1647774728&context=system.io&dimensions=out&format=json&group=average&gtime=0&options=absolute%7Callow_past%7Cflip%7Cjsonwrap%7Cms&points=40" not found! Returning 404 to Cloud!
```